test/acceptance/features/steps/steps.py

```diff
@@ -466,3 +468,10 @@ def validate_secret_empty(context, secret_name):
                       ignore_exceptions=(json.JSONDecodeError,))
     except polling2.TimeoutException:
         pass
+
+
+@given(u'"{app_name}" is deployed from image "{image_ref}"')
+def create_deployment(context, app_name, image_ref):
+    cmd = Command()
+    output, exit_code = cmd.run(f'kubectl create deployment --image={image_ref} {app_name}')
+    assert exit_code == 0, f"Could not create deployment - {output}"
```

Comment:
We already have `App` class that models any application:

https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/steps/app.py#L6

Let's reuse it.

Reply:
In this function its also exposing application but in our example which we're verifying here, we're not. Does splitting these two tasks(deploying the app and exposing it) make sense?

Reply:
Exposure of a deployment only creates a Service resource that can be used to access the application. Which is important in case of the generic test `App` to be able to access the API by tests.
In case of the example, only the `Deployment` while no API is accessed, at all. However, the exposure is just additional layer that does not interfere with the example in any matter (since the example only access the intermediate secret). So use of the generic application actually implements the example's step.

Reply:
Okay, I'll update implementation.

Reply:
@pedjak PTAL.